### DISCUSSION TITLE
Add hook when Consumer loses connection to RMQ

### DISCUSSION THREAD

**Discussion by peter-a** [rabbitmq/rabbitmq-stream-java-client#290] [Q&A] [answered]
If RMQ goes down for some reason, I need to provide some fallback logic. Currently there is no hook for this scenario. I tried implementing https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/main/src/main/java/com/rabbitmq/stream/ConsumerUpdateListener.java but this listener is not called when RMQ is down.

**Comment by acogoluegnes** [comment, marked as answer]
Can you please tell us more about the fallback logic you're mentioning?

The client library takes care of reconnecting and re-enabling producer and consumer instances.
